tests/unit/test-api.c

```diff
@@ -1002,6 +1017,28 @@ main (void)
     jerry_cleanup ();
   }
 
+  /* Test stopping an infinite loop. */
+  if (jerry_is_feature_enabled (JERRY_FEATURE_VM_EXEC_STOP))
+  {
+    jerry_init (JERRY_INIT_EMPTY);
+    const char *inf_loop_code_src_p = "while(true) {}";
+    parsed_code_val = jerry_parse ((jerry_char_t *) inf_loop_code_src_p, strlen (inf_loop_code_src_p), false);
+    TEST_ASSERT (!jerry_value_has_error_flag (parsed_code_val));
+
+    int countdown = 6;
+
+    jerry_set_vm_exec_stop_callback (vm_exec_stop_callback, &countdown, 16);
```

Comment:
To be more precise it counts the backward branches, function entries and stack rollbacks, but in your case there is only a single function entry (the global code is a function). 

This code stops sooner:

    for (i = 0; i<6*16; i++)
    {
      try {
       throw 1
      } catch(e) {
      }
    }